tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/BaseLinker.scala

```diff
@@ -115,7 +116,18 @@ final class BaseLinker(semantics: Semantics, esLevel: ESLevel, considerPositions
       }
 
       val linkingErrLevel = if (fatal) Level.Error else Level.Warn
-      analysis.errors.foreach(logError(_, logger, linkingErrLevel))
+      val maxDisplayErrors = {
+        val propName = "org.scalajs.core.tools.linker.maxlinkingerrors"
+        Try(System.getProperty(propName, "20").toInt).getOrElse(20).max(1)
```

Comment:
> How is this not a global, mutable, untyped variable?

It's *slightly* not as bad as that. System properties are not *supposed* to be mutated. An almost reasonable way to use them is to set them once at "process" creation. But yes, it's definitely global, untyped, and potentially mutable.

The alternative was not to offer any means to configure this thing. Yes, we could add a new options object for `BaseLinker` for this option, but then we'd have to trail that option through the layer of abstractions of the linker that we have. Unlike all the other things we allow to configure, this one doesn't even have a use case. Complicating the interface through several layers for something that will never be set, and only has a "visual" impact for a human, is not worth it.

Another possible way to configure it would be a `DynamicVariable` in the companion object of `BaseLinker`. Still global, still mutable, but at least thread-safe and typed. But, for a human, harder to configure.